test/e2e_new/single_identity_test.go

```diff
@@ -147,23 +150,79 @@ var _ = Describe("[PR] When deploying one identity", func() {
 		})
 	})
 
-	It("should establish a new AzureAssignedIdentity and remove the old one when draining the node containing identity validator", func() {
-
-	})
+	It("should update AzureAssignedIdentity and when AzureIdentity fields are updated", func() {
+		azureIdentity = azureidentity.Update(azureidentity.UpdateInput{
+			Updater:             kubeClient,
+			Config:              config,
+			AzureClient:         azureClient,
+			AzureIdentity:       azureIdentity,
+			UpdatedIdentityName: fmt.Sprintf("%s-0", keyvaultIdentity),
+		})
 
-	It("should pass liveness probe test", func() {
+		azureassignedidentity.WaitForLen(azureassignedidentity.WaitForLenInput{
```

Comment:
The `AzureAssignedIdentity` is already 1 at this point. So when we validate it's possible MIC is actually assigning the new identity and we'll get an error `"identity not found"`. The status is flipped to `Created` right before the update, so we should also check if the `AzureAssignedIdentity` state is `Assigned` before validating